### PR TITLE
adding basic support for subPackages

### PR DESCRIPTION
**Pull request by Jimexist**
adding basic support for subPackages.

screen shot 2018-09-11 at 12 00 05

### CODE REVIEW

**Comment by Jimexist** [comment]
@imyelo do you know how to handle the _actual_ resolving? i can only get the json transformation done then i don't have that much time to investigate

File: packages/mina-entry-webpack-plugin/index.js [hunk 1]
```diff
@@ -39,12 +39,23 @@ function getUrlsFromConfig(config) {
...
 
-  ;['pages', 'usingComponents', 'publicComponents'].forEach(prop => {
-    if (typeof config[prop] !== 'object') {
-      return
-    }
+  let components = ['pages', 'usingComponents', 'publicComponents'].map(
+    c => config[c]
+  )
+
+  if (Array.isArray(config['subPages'])) {
```
Comment (imyelo) [inline packages/mina-entry-webpack-plugin/index.js]: should be `subPackages`

File: packages/mina-entry-webpack-plugin/index.js [hunk 2]
```diff
@@ -39,12 +39,23 @@ function getUrlsFromConfig(config) {
...
+    config['subPages'].forEach(subPage => {
+      const { root, pages } = subPage
+      if (Array.isArray(pages)) {
+        pages.forEach(page => {
```
Comment (imyelo) [inline packages/mina-entry-webpack-plugin/index.js]: I think it should be ``components = [...components, pages.map(page => path.join(root, page))]``

File: packages/mina-entry-webpack-plugin/index.js [hunk 3]
```diff
@@ -32,21 +40,28 @@ function readConfig(fullpath) {
 
 function getUrlsFromConfig(config) {
   let urls = []
+
   if (!config) {
     return urls
   }
-  if (Array.isArray(config.pages)) {
-    urls = [...urls, ...config.pages]
```
[the remark this answers is not in the document]
Comment (imyelo) [reply packages/mina-entry-webpack-plugin/index.js]: @lizheming 因为发现在后面当作对象处理时，`values` 方法 (也就是 `Object.keys()` + `.map`) 也覆盖到了是数组的情况

[the remark this answers is not in the document]
Comment (lizheming) [reply packages/mina-entry-webpack-plugin/index.js]: @imyelo OK，了解。

[the remark this answers is not in the document]
Comment (imyelo) [reply packages/mina-entry-webpack-plugin/index.js]: 哈 我应该单独提个 commit 的

[the remark this answers is not in the document]
Comment (lizheming) [reply packages/mina-entry-webpack-plugin/index.js]: @imyelo 没关系，是我进了自己的套子了，之前一直是记得后面是处理对象的逻辑，所以人肉过代码的时候数组就自动跳过去了OwQ

**Pull request merged**